File: launchdarkly/java/src/main/java/oseg/launchdarkly_examples/CreateFlagLinkExample.java
package oseg.launchdarkly_examples;

import com.launchdarkly.client.ApiException;
import com.launchdarkly.client.Configuration;
import com.launchdarkly.client.api.*;
import com.launchdarkly.client.auth.*;
import com.launchdarkly.client.JSON;
import com.launchdarkly.client.model.*;

import java.io.File;
import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.OffsetDateTime;
import java.util.ArrayList;
import java.util.List;
import java.util.Map;

public class CreateFlagLinkExample
{
    public static void main(String[] args)
    {
        var config = Configuration.getDefaultApiClient();
        ((ApiKeyAuth) config.getAuthentication("ApiKey")).setApiKey("YOUR_API_KEY");

        var flagLinkPost = new FlagLinkPost();
        flagLinkPost.key("flag-link-key-123abc");
        flagLinkPost.deepLink("https://example.com/archives/123123123");
        flagLinkPost.title("Example link title");
        flagLinkPost.description("Example link description");

        try
        {
            var response = new FlagLinksBetaApi(config).createFlagLink(
                "projectKey_string", // projectKey
                "featureFlagKey_string", // featureFlagKey
                flagLinkPost
            );

            System.out.println(response);
        } catch (ApiException e) {
            System.err.println("Exception when calling FlagLinksBetaApi#createFlagLink");
            System.err.println("Status code: " + e.getCode());
            System.err.println("Reason: " + e.getResponseBody());
            System.err.println("Response headers: " + e.getResponseHeaders());
            e.printStackTrace();
        }
    }
}

File: launchdarkly/kotlin/src/main/kotlin/oseg/launchdarkly_examples/CreateFlagLinkExample.kt
package oseg.launchdarkly_examples

import com.launchdarkly.client.infrastructure.*
import com.launchdarkly.client.apis.*
import com.launchdarkly.client.models.*

import java.io.File
import java.time.LocalDate
import java.time.OffsetDateTime
import kotlin.collections.ArrayList
import kotlin.collections.List
import kotlin.collections.Map
import com.squareup.moshi.adapter

@ExperimentalStdlibApi
class CreateFlagLinkExample
{
    fun createFlagLink()
    {
        ApiClient.apiKey["Authorization"] = "YOUR_API_KEY"

        val flagLinkPost = FlagLinkPost(
            key = "flag-link-key-123abc",
            deepLink = "https://example.com/archives/123123123",
            title = "Example link title",
            description = "Example link description",
        )

        try
        {
            val response = FlagLinksBetaApi().createFlagLink(
                projectKey = null,
                featureFlagKey = null,
                flagLinkPost = flagLinkPost,
            )

            println(response)
        } catch (e: ClientException) {
            println("4xx response calling FlagLinksBetaApi#createFlagLink")
            e.printStackTrace()
        } catch (e: ServerException) {
            println("5xx response calling FlagLinksBetaApi#createFlagLink")
            e.printStackTrace()
        }
    }
}
